examples/univprop.ctt

```diff
@@ -0,0 +1,707 @@
+
+Fforgetful : functor Ab CMon
+  = let
+    f : cgroup -> cmonoid
+      = forgetfulAb
+    Fmor (x y : cgroup) (g : cgrouphom x y) : (cmonoidhom (f x) (f y))
+      = g
+    Fid (x : cgroup)
+      : Path (cmonoidhom (f x) (f x)) (Fmor x x (idmonoidhom (x.1, x.2.1.1))) (idmonoidhom ((f x).1, (f x).2.1))
+      = <_> idmonoidhom ((f x).1, (f x).2.1)
+    p (x y z : cgroup) (g : cgrouphom x y) (h : cgrouphom y z) 
+      : Path (cmonoidhom (f x) (f z))
+          (Fmor x z (monoidhomcomp (x.1, x.2.1.1) (y.1, y.2.1.1) (z.1, z.2.1.1) g h))
+          (monoidhomcomp ((f x).1, (f x).2.1) ((f y).1, (f y).2.1) ((f z).1, (f z).2.1)
+            (Fmor x y g) (Fmor y z h))
+      = <_> Fmor x z (monoidhomcomp (x.1, x.2.1.1) (y.1, y.2.1.1) (z.1, z.2.1.1) g h)
+  in (f, Fmor, Fid, p)
+
+Fgrothendieck : functor CMon Ab
+  = let
+    f : cmonoid -> cgroup
+      = grothendieck
+    Fmor (x y : cmonoid) (g : cmonoidhom x y) : (cgrouphom (f x) (f y))
+      = grothendiecklem4 x (f y)
+          (monoidhomcomp (x.1, x.2.1) (y.1, y.2.1) ((f y).1, (f y).2.1.1) g (grothendieckhom y))
+    Fid (x : cmonoid)
+      : Path (cgrouphom (f x) (f x)) (Fmor x x (idmonoidhom (x.1, x.2.1))) (idmonoidhom ((f x).1, (f x).2.1.1))
+      = let
+        i : monoidhom (x.1, x.2.1) (x.1, x.2.1)
+          = idmonoidhom (x.1, x.2.1)
+        a : cmoncgrouphom x (f x)
+          = monoidhomcomp (x.1, x.2.1) (x.1, x.2.1) ((f x).1, (f x).2.1.1) i (grothendieckhom x)
+        g : cgrouphom (f x) (f x)
+          = grothendiecklem4 x (f x) a
+        h : cgrouphom (f x) (f x)
+          = idmonoidhom ((f x).1, (f x).2.1.1)
+        P (z : (f x).1.1) : hProp
+          = ( Path (f x).1.1 (g.1 z) (h.1 z)
+            , (f x).1.2 (g.1 z) (h.1 z) )
+        ps (z : pair x) : Path (f x).1.1 (g.1 (qpr x z)) (qpr x z)
+          = <i> comp (<_> (f x).1.1) (qopeq x (z.1, id x) (id x, z.2) @ i)
+            [ (i = 0) -> <j> comp (<_> (f x).1.1) (qop x (qpr x (z.1, id x)) (qinveq x (z.2, id x) @ -j))
+              [ (j = 0) -> <_> qop x (qpr x (z.1, id x)) (qpr x (id x, z.2))
+              , (j = 1) -> <k> grothendiecklem3 x (f x) a z @ -k ]
+            , (i = 1) -> <j> qpr x ((hid x).2 z.1 @ j, (hid x).1 z.2 @ j) ]
+        p : Path ((f x).1.1 -> (f x).1.1) g.1 h.1
+          = <i> \ (z : (f x).1.1) -> (setquotunivprop (pair x) (peqrel x) P ps z @ i)
+      in lemSig ((f x).1.1 -> (f x).1.1)
+           (ismonoidhom ((f x).1, (f x).2.1.1) ((f x).1, (f x).2.1.1))
+           (propismonoidhom ((f x).1, (f x).2.1.1) ((f x).1, (f x).2.1.1))
+           g h p
+    -- A bit too slow to compile
+    p (x y z : cmonoid) (g : cmonoidhom x y) (h : cmonoidhom y z)
+      : Path (cgrouphom (f x) (f z))
+          (Fmor x z (monoidhomcomp (x.1, x.2.1) (y.1, y.2.1) (z.1, z.2.1) g h))
+          (monoidhomcomp ((f x).1, (f x).2.1.1) ((f y).1, (f y).2.1.1) ((f z).1, (f z).2.1.1)
+            (Fmor x y g) (Fmor y z h))
+      = undefined
```

Comment:
Ok, that's slow but not extremely slow. Let's remove the `undefined` so that the commented code doesn't break if something changes.